extensions/ql-vscode/src/remote-queries/variant-analysis-manager.ts

```diff
@@ -224,6 +224,18 @@ export class VariantAnalysisManager extends DisposableObject implements VariantA
     await createTimestampFile(this.getVariantAnalysisStorageLocation(variantAnalysisId));
   }
 
+  public async openVariantAnalysisResults(variantAnalysisId: number): Promise<void> {
+    try {
+      const variantAnalysis = this.variantAnalyses.get(variantAnalysisId);
+      if (!variantAnalysis) {
+        void showAndLogErrorMessage(`No variant analysis with id: ${variantAnalysisId}`);
+      }
+      void commands.executeCommand('codeQL.openVariantAnalysisView', variantAnalysisId);
```

Comment:
Is there a reason we're not calling `this.showView()` directly instead of executing a command?

Also, since there is a command to show a view, I'm wondering if it makes sense to execute that command directly from query history, rather than calling a function on the `VariantAnalysisManager`. What do you think? 

We could add some of the nice things you added around defensiveness to the `showView()` function.

Reply:
> Is there a reason we're not calling `this.showView()` directly instead of executing a command?

No good reason 😅 I was following the example of [`openRemoteQueryResults`](https://github.com/github/vscode-codeql/blob/c26d786a1cac98ba4f6bf22e65133fdbba60e38d/extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts#L103), but I agree that it would be more sensible to just call `showView` directly! 

Some of that defensiveness [already exists in the view](https://github.com/github/vscode-codeql/blob/6a636baa2139a7fac8e706d98feafce9bc41233e/extensions/ql-vscode/src/remote-queries/variant-analysis-view.ts#L109-L112) anyway, so I definitely didn't need to duplicate it 🤦🏽 

Clean-up incoming! 🧹